src/rpc/cache.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 The Pocketcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef POCKETCOIN_RPC_CACHE_H
+#define POCKETCOIN_RPC_CACHE_H
+#include <univalue.h>
+#include <sync.h>
+#include <logging.h>
+#include <validation.h>
+
+class JSONRPCRequest;
+
+class RPCCache
+{
+private:
+    Mutex CacheMutex;
+    std::map<std::string, UniValue> m_cache;
+    int m_blockHeight;
+    int m_cacheSize;
+    int m_maxCacheSize;
+    const std::vector<std::string> supportedMethods = {"getlastcomments",
```

Comment:
Please expand the list of cached methods:
```
getlastcomments
getcomments
getuseraddress
getaddressregistration
search
searchlinks
searchusers
gettags
getrawtransactionwithmessagebyid
getrawtransactionwithmessage
getrawtransaction
getusercontents
gethistoricalfeed
gethistoricalstrip
gethierarchicalfeed
gethierarchicalstrip
gethotposts
getuserprofile
getprofilefeed
getsubscribesfeed
txunspent
getaddressid
getuserstate
getpagescores
getcontent
getcontents
getaccountsetting
getcontentsstatistic
getusersubscribes
getusersubscribers
getuserblockings
getaddressscores
getpostscores
getpagescores
getstatisticbyhours
getstatisticbydays
getstatisticcontentbyhours
getstatisticcontentbydays
getrecomendedaccountsbytags
getrecomendedcontentsbyscoresonsimilarcontents
getaddressinfo
getcompactblock
getlastblocks
searchbyhash
gettransactions
getaddresstransactions
getblocktransactions
getbalancehistory
getcoininfo
estimatesmartfee
```

Reply:
Added a new commit to this PR with the updated list of RPC commands.